src/Serilog/Util/TimeProvider.cs

```diff
@@ -0,0 +1,48 @@
+// Copyright © Serilog Contributors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#if !NET8_0_OR_GREATER
+
+using System.Diagnostics;
+
+namespace System;
+
+/// <summary>
+/// A super-simple, cut-down subset of `System.TimeProvider` which we use internally to avoid a package dependency
+/// on platforms without it.
+/// </summary>
+abstract class TimeProvider
```

Comment:
Adding a package dependency to pull this in on downlevel platforms would be a significant change that we'd hold for a major version bump. The minimal subset of `System.TimeProvider` implemented here should at least get us started with integrating it in a few places.